modules/ROOT/pages/_partials/commons/common-indexing.adoc

```diff
@@ -109,6 +109,322 @@ include::{root-partials}block_tabbed_code_example.adoc[]
+
+Below is an example array index configuration:
+
+ifdef::is-android[]
+
+[tabs]
+=====
+Java::
++
+--
+[source, java]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/ArrayIndexExamples.java[tags=array-index-config]
+
+----
+--
+
+Kotlin::
++
+--
+[source, kotlin]
+----
+
+include::android:example$kotlin_snippets/app/src/main/kotlin/com/couchbase/codesnippets/ArrayIndexExamples.kt[tags=array-index-config]
+
+----
+--
+=====
+
+endif::is-android[]
+ifdef::is-java[]
+
+[source, java]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/ArrayIndexExamples.java[tags=array-index-config]
+
+----
+
+endif::is-java[]
+ifdef::is-csharp[]
+
+[source, csharp]
+----
+include::csharp:example$code_snippets/Program.cs[tags=array-index-config]
+----
+endif::is-csharp[]
+ifdef::is-objc[]
+[source, objc]
+----
+
+include::objc:example$code_snippets/VectorSearch.m[tags=array-index-config]
+
+----
+endif::is-objc[]
+ifdef::is-swift[]
+[source, swift]
+----
+
+include::swift:example$code_snippets/VectorSearch.swift[tags=array-index-config]
+
+----
+endif::is-swift[]
+ifdef::is-c[]
+[source, c]
+----
+
+include::c:example$code_snippets/main.cpp[tags=array-index-config]
+
+----
+endif::is-c[]
+
+=== Array Index Syntax
+
+The syntax for array index configuration is shown below:
+
+[cols ="3*"]
+|===
+|Name |Is Optional? |Description
+
+|`path`
+|image:ROOT:no.png[]
+|Path to the array to be indexed.
+
+|`expressions`
+|image:ROOT:yes.png[]
+|An optional array of strings, where each string represents an expression defining the values within the array to be indexed.
+Use `"[]"` to represent a property that is an array of each nested array level.
+For instance `contacts[].phones` to specify an array of phones within each contact.
+For a single array or the last level array, the `"[]"` is optional. 
+If the array specified by the path contains scalar values, this parameter can be null.
+
+|===
+
+=== Using Array Indexes with UNNEST
+
+For the following examples, you can assume we are querying results from the following document, shown below:
+
+[source, JSON]
+--
+{
+   "Name":"Sam",
+   "contacts":[
+     {
+       "type":"primary",
+       "address":{"street":"1 St","city":"San Pedro","state":"CA"},
+       "phones":[
+         {"type":"home","number":"310-123-4567"},
+         {"type":"mobile","number":"310-123-6789"}
+       ]
+     },
+     {
+       "type":"secondary",
+       "address":{"street":"5 St","city":"Seattle","state":"WA"},
+       "phones":[
+         {"type":"home","number":"206-123-4567"},
+         {"type":"mobile","number":"206-123-6789"}
+       ]
+     }
+   ],
+   "likes":["soccer","travel"]
+ }
+
+--
+
+Using the document above we can perform queries on a single nested array like so:
+
+[source, sqlpp]
+--
+SELECT name, like FROM _ UNNEST likes as like WHERE like = "Travel" 
```

Comment:
@ElliotFrancisHunter this applies to all of these examples across platforms but what is the underscore? Shouldn't that be the name of the data source you are querying?

Reply:
I will add a brief preamble about this. It's shorthand notation for the current database you're accessing.